x-pack/plugins/infra/public/alerting/log_threshold/components/alert_details_app_section/components/explain_log_rate_spikes.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import { pick } from 'lodash';
+import moment from 'moment';
+
+import { EuiFlexGroup, EuiFlexItem, EuiPanel, EuiTitle } from '@elastic/eui';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { ExplainLogRateSpikesContent } from '@kbn/aiops-plugin/public';
+
+import { Rule } from '@kbn/alerting-plugin/common';
+import { TopAlert } from '@kbn/observability-plugin/public';
+import { estypes } from '@elastic/elasticsearch';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { useKibanaContextForPlugin } from '../../../../../hooks/use_kibana';
+import {
+  CountRuleParams,
+  PartialRuleParams,
+  ruleParamsRT,
+} from '../../../../../../common/alerting/logs/log_threshold';
+import { DEFAULT_LOG_VIEW } from '../../../../../observability_logs/log_view_state/src/defaults';
+import { getESQuery } from '../query';
+import { decodeOrThrow } from '../../../../../../common/runtime_types';
+
+export interface AlertDetailsExplainLogRateSpikesSectionProps {
+  rule: Rule<PartialRuleParams>;
+  alert: TopAlert<Record<string, any>>;
+}
+
+export const ExplainLogRateSpikes: FC<AlertDetailsExplainLogRateSpikesSectionProps> = ({
+  rule,
+  alert,
+}) => {
+  const { services } = useKibanaContextForPlugin();
+  const { dataViews, logViews } = services;
+  const [dataView, setDataView] = useState<DataView | undefined>();
+  const [esSearchQuery, setEsSearchQuery] = useState<QueryDslQueryContainer | undefined>();
+
+  useEffect(() => {
+    async function getDataView() {
+      const { indices, timestampField, runtimeMappings, dataViewReference } =
+        await logViews.client.getResolvedLogView(DEFAULT_LOG_VIEW);
+      const dataViewId = dataViewReference.id;
+
+      if (dataViewId) {
+        const logDataView = await dataViews.get(dataViewId);
+        setDataView(logDataView);
+
+        const esSearchRequest = getESQuery(
+          decodeOrThrow(ruleParamsRT)(rule.params) as CountRuleParams,
+          timestampField,
+          indices,
+          runtimeMappings,
+          alert.start
+        ) as estypes.SearchRequest;
+
+        if (!esSearchRequest) {
+          throw new Error('ES query could not be built from the provided alert params');
+        }
+
+        setEsSearchQuery(esSearchRequest.query);
+      }
+    }
+
+    getDataView();
+  }, [rule, alert, dataViews, logViews]);
+
+  const timeRange = { min: moment(alert.start).subtract(20, 'minutes'), max: moment(new Date()) };
```

Comment:
If an alert has been resolved for a while, this will still use `now` for the end of the time range for Explain Log Rate Spikes. After some time this will make it harder for the tool to pick the right analysis time range, for example:

<img width="1068" alt="image" src="https://github.com/elastic/kibana/assets/230104/5e1d16ac-b902-45b6-bb47-4e17c1e36ce8">


I had a go at some code that uses the alert's start and end if available, additionally this will choose the middle point of the alert's start and end point for the initial analysis start. We could further improve this based on the available data but it will need some update to the <ExplainLogRateSpikesContent>, I'll continue iterating on this:

```js
const kibanaAlertStart = moment(alert.fields['kibana.alert.start'] ?? alert.start);
  const kibanaAlertEnd = moment(alert.fields['kibana.alert.end'] ?? new Date());

  const timeRange = {
    min: kibanaAlertStart.clone().subtract(20, 'minutes'),
    max: alert.fields['kibana.alert.end']
      ? kibanaAlertEnd.clone().add(20, 'minutes')
      : kibanaAlertEnd.clone(),
  };

  // Analysis start timestamp should be the middle point of the start and end of the alert.
  const initialAnalysisStart = kibanaAlertStart
    .clone()
    .add(kibanaAlertEnd.clone().diff(kibanaAlertStart) / 2)
    .valueOf();
```

Here's how the updated time range and initial analysis start point looks: 

<img width="1076" alt="image" src="https://github.com/elastic/kibana/assets/230104/b34a73e6-a48f-4f66-b85f-f278e311c466">

Reply:
Yes, I agree it would be good to use `alert.start` - we need to adapt the props of `ExplainLogRateSpikesContent` a bit to make that possible, at the moment we pass in `alert.start` but then the analysis will auto-choose a range left and right of that start.

Reply:
I see what you mean. I think we need to fix the deviation start time for initial analysis. Deviation start could be alert start time and deviation end would be "some time" after alert start.

Another issue is, if the log rate is already above threshold when the rule is created, then there may not be much difference between baseline and deviation.

If we can somehow look at the data distribution, and calculate baseline and deviation based on that, that would be great. How are the baseline and deviation calculated currently, is it based on the timestamps or data? If we can base  these on the data, we then need to adjust time range to sufficiently find baseline.

Reply:
I created a PR that will allow us to pass in more fine grained parameters to set the baseline and deviation time ranges based on the alert metadata: https://github.com/elastic/kibana/pull/159111

At the moment it's purely based on timeranges relative to the passed in initial timestamp (`alert.start`), it's described in more detail here: https://github.com/elastic/kibana/blob/main/x-pack/packages/ml/aiops_utils/src/get_window_parameters.ts#L55

Once the above PR is merge, we're able to do something like this:

```js
  const alertStart = moment(alert.start);
  const initialAnalysisStart = {
    baselineMin: alertStart.subtract(15, 'minutes').valueOf(),
    baselineMax: alertStart.subtract(1, 'minutes').valueOf(),
    deviationMin: alert.start,
    deviationMax: alertStart.add(5, 'minutes').valueOf(),
  };
```

<img width="1076" alt="image" src="https://github.com/elastic/kibana/assets/230104/4d32515a-ce7f-45e9-b3f5-b77117c3286d">

To be able to set baseline/deviation based on data would be great at some point, but it will need a bit more work.

Reply:
#159111 has been merged so the example above with passing in baseline+deviation time ranges should be possible from now on.